### DISCUSSION TITLE
[feat]: conventions for unnamed imports / component types / icons use case

### DISCUSSION THREAD

**Discussion by med8bra** [shadcn-ui/ui#5751] [Ideas]
### Feature description

Provide possibility to customize imports naming for some type of components. I find my self doing multiple search/rename when adding a component to accommodate my preferred  conventions, example with icons:
- I prefer to have an extra `Icon` suffix in icon component:
```typescript
// before
import { X, Search } from "lucide-react";
// after (lucide supports these imports too, instead of renaming)
import { XIcon, SearchIcon } from "lucide-react";
```  

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues and PRs

**Comment by shadcn** [comment]
Agreed on this. It's on my list to add.

**Comment by med8bra** [comment]
@shadcn do you have any ideas on how you would approach this? 

I just did a quick read over the code base, I see that all components/blocks are created manually and they are the final artifact I believe. So there is no processing done to a component like using a templating language.

I like that the code base is very simple in terms of how the components are made and shipped, and better to keep this simplicity. 

Some ideas that may help with this need:
- Post processing components on client side: Using codemods for example to apply a naming convention (that's what I'm using currenlty for lucide). There could be an abstraction layer for the rules, and the implementation could be via  [jscodeshift](https://jscodeshift.com)
- Using a templating language for component, but I'm not sure about this approach.
